code/game/mecha/mecha.dm

```diff
@@ -2018,26 +2079,46 @@
 		icon_state = initial(icon_state)
 	return icon_state
 
+//This is for mobs mostly.
 /obj/mecha/attack_generic(var/mob/user, var/damage, var/attack_message)
 
 	user.setClickCooldown(user.get_attack_speed())
 	if(!damage)
 		return 0
 
 	src.log_message("Attacked. Attacker - [user].",1)
-
 	user.do_attack_animation(src)
-	if(!prob(src.deflect_chance))
-		src.take_damage(damage)
-		src.check_for_internal_damage(list(MECHA_INT_TEMP_CONTROL,MECHA_INT_TANK_BREACH,MECHA_INT_CONTROL_LOST))
-		visible_message("<span class='danger'>[user] [attack_message] [src]!</span>")
-		user.attack_log += text("\[[time_stamp()]\] <font color='red'>attacked [src.name]</font>")
-	else
+
+	//var/pass_damage	//See the comment in the larger greyed out block below.
+	//var/pass_damage_reduc_mod
+	if(prob(src.deflect_chance))//Deflected
 		src.log_append_to_last("Armor saved.")
-		playsound(src, 'sound/weapons/slash.ogg', 50, 1, -1)
 		src.occupant_message("<span class='notice'>\The [user]'s attack is stopped by the armor.</span>")
 		visible_message("<span class='notice'>\The [user] rebounds off [src.name]'s armor!</span>")
 		user.attack_log += text("\[[time_stamp()]\] <font color='red'>attacked [src.name]</font>")
+		playsound(src, 'sound/weapons/slash.ogg', 50, 1, -1)
+
+	else if(damage < damage_minimum)//Pathetic damage levels just don't harm MECH.
+		src.occupant_message("<span class='notice'>\The [user]'s doesn't dent \the [src] paint.</span>")

```

Comment:
As above, so below.

Reply:
Changed. 
For below, it actually does a fair amount of damage, so the deflect would be misleading.